protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -2328,6 +2385,12 @@ func updateswaggerObjectFromJSONSchema(s *openapiSchemaObject, j *openapi_option
 	s.MaxProperties = j.GetMaxProperties()
 	s.MinProperties = j.GetMinProperties()
 	s.Required = j.GetRequired()
+	if reg.GetUseJSONNamesForFields() {
+		for i, r := range s.Required {
+			// TODO(oyvindwe): Look up field and use field.GetJsonName()?
+			s.Required[i] = doCamelCase(r)
```

Comment:
This makes me a little nervous, if we could look up the field and use the canonical field.GetJsonName that would be much preferred, but if it's too much trouble we can leave it for a follow up.

Reply:
It is a bug in the the existing code that just passes the `required` array on without transforming to the JSON format as you can see here: https://github.com/grpc-ecosystem/grpc-gateway/blob/09a279d441d5819789b08c0ae17544045601ca67/examples/internal/proto/examplepb/a_bit_of_everything.swagger.json#L2091 and here: https://github.com/grpc-ecosystem/grpc-gateway/blob/09a279d441d5819789b08c0ae17544045601ca67/examples/internal/proto/examplepb/a_bit_of_everything.swagger.json#L3744

This also broke my new check to set the boolean `required` flag in this PR.

I investigated fixing this properly, but a lot of context is missing here to look up field name. I considered transforming the `Required` slice to JSON format earlier, but that would require different check vs. field names. I think the best approach is to represent `Required` as fields rather than strings, and look them up when retrieving the options from the proto file (and discard/warn about undefined fields), as with this annotation it's possible to generate illegal an OpenAPIv2 file. I'd consider this outside the scope of this PR.

Reply:
That is fair, lets leave this for now.